lib/node_modules/@stdlib/blas/base/wasm/sdsdot/docs/types/test.ts

```diff
@@ -0,0 +1,577 @@
+
+	mod.main( 10, 1, 0, 1, 80, '10' ); // $ExpectError
+	mod.main( 10, 1, 0, 1, 80, true ); // $ExpectError
+	mod.main( 10, 1, 0, 1, 80, false ); // $ExpectError
+	mod.main( 10, 1, 0, 1, 80, null ); // $ExpectError
+	mod.main( 10, 1, 0, 1, 80, undefined ); // $ExpectError
+	mod.main( 10, 1, 0, 1, 80, [] ); // $ExpectError
+	mod.main( 10, 1, 0, 1, 80, {} ); // $ExpectError
+	mod.main( 10, 1, 0, 1, 80, ( x: number ): number => x ); // $ExpectError
+}
+
+// The compiler throws an error if the `main` method of a module instance is provided an unsupported number of arguments...
+{
+	const mem = new Memory({
+		'initial': 1
+	});
+	const mod = sdsdot.Module( mem );
+
+	mod.main(); // $ExpectError
+	mod.main( 10 ); // $ExpectError
+	mod.main( 10, 1 ); // $ExpectError
+	mod.main( 10, 1,0 ); // $ExpectError
+	mod.main( 10, 1, 0, 1 ); // $ExpectError
+	mod.main( 10, 1, 0, 1, 80 ); // $ExpectError
+	mod.main( 10, 1, 0, 1, 80, 1, 10 ); // $ExpectError
+}
+
+// The `Module` constructor returns a module instance having an `ndarray` method which returns a number...
+{
+	const mem = new Memory({
+		'initial': 1
+	});
+	const mod = sdsdot.Module( mem );
+
+	mod.ndarray( 10, 0, 0, 1, 0, 80, 1, 0 ); // $ExpectType number
+}
+
+// The compiler throws an error if the `ndarray` method of a module instance is provided a first argument which is not a number...
+{
+	const mem = new Memory({
+		'initial': 1
+	});
+	const mod = sdsdot.Module( mem );
+
+	mod.ndarray( '10', 0, 0, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( true, 0, 0, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( false, 0, 0, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( null, 0, 0, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( undefined, 0, 0, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( [], 0, 0, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( {}, 0, 0, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( ( x: number ): number => x, 0, 0, 1, 0, 80, 1, 0 ); // $ExpectError
+}
+
+// The compiler throws an error if the `ndarray` method of a module instance is provided a second argument which is not a number...
+{
+	const mem = new Memory({
+		'initial': 1
+	});
+	const mod = sdsdot.Module( mem );
+
+	mod.ndarray( 10, '10', 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( 10, true, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( 10, false, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( 10, null, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( 10, undefined, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( 10, [], 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( 10, {}, 1, 0, 80, 1, 0 ); // $ExpectError
+	mod.ndarray( 10, ( x: number ): number => x, x, 1, 0, 80, 1, 0 ); // $ExpectError
+}
+
+// The compiler throws an error if the `ndarray` method of a module instance is provided a third argument which is not a number...
+{
+	const mem = new Memory({
+		'initial': 1
+	});
+	const mod = sdsdot.Module( mem );
+
+	mod.ndarray( 10, 0, '10', 0, 80, 1, 0 ); // $ExpectError
```

Comment:
You're missing an argument. You are only providing seven arguments and there are eight.